src/borrower/feed/navfeedPV.sol

```diff
@@ -0,0 +1,222 @@
+        for (uint loanId = 1; loanId <= shelf.loanCount(); loanId++) {
+            totalDebt = safeAdd(totalDebt, pile.debt(loanId));
+        }
+
+        // substract writtenoff loans -> all writtenOff loans are moved to writeOffRateGroup
+        totalDebt = safeSub(totalDebt, pile.rateDebt(WRITEOFF_RATE_GROUP));
+        return totalDebt;
+    }
+
+    function calcUpdateNAV() public returns(uint) {
+        latestNAV = currentNAV();
+        lastNAVUpdate = uniqueDayTimestamp(block.timestamp);
+        return latestNAV;
+    }
+
+    function update(bytes32 nftID_,  uint value) public auth {
+        // switch of collateral risk group results in new: ceiling, threshold for existing loan
+        details[nftID_].nftValues = toUint128(value);
+        emit Update(nftID_, value);
+    }
+
+    function update(bytes32 nftID_, uint value, uint risk_) public auth {
+        uint nnow = uniqueDayTimestamp(block.timestamp);
+        details[nftID_].nftValues  = toUint128(value);
+
+        // no change in risk group
+        if (risk_ == risk(nftID_)) {
+            return;
+        }
+
+        // nfts can only be added to risk groups that are part of the score card
+        require(thresholdRatio(risk_) != 0, "risk group not defined in contract");
+        details[nftID_].risk = toUint128(risk_);
+
+        // switch of collateral risk group results in new: ceiling, threshold and interest rate for existing loan
+        // change to new rate interestRate immediately in pile if loan debt exists
+        uint loan = shelf.nftlookup(nftID_);
+        if (pile.pie(loan) != 0) {
+            pile.changeRate(loan, risk_);
+        }
+        emit Update(nftID_, value, risk_);
+    }
+
+    // --- Utilities ---
+    // returns the threshold of a loan
+    // if the loan debt is above the loan threshold the NFT can be seized
+    function threshold(uint loan) public view returns (uint) {
+        bytes32 nftID_ = nftID(loan);
+        return rmul(nftValues(nftID_), thresholdRatio(risk(nftID_)));
+    }
+
+    // returns a unique id based on the nft registry and tokenId
+    // the nftID is used to set the risk group and value for nfts
+    function nftID(address registry, uint tokenId) public pure returns (bytes32) {
+        return keccak256(abi.encodePacked(registry, tokenId));
+    }
+
+    // returns the nftID for the underlying collateral nft
+    function nftID(uint loan) public view returns (bytes32) {
+        (address registry, uint tokenId) = shelf.shelf(loan);
+        return nftID(registry, tokenId);
+    }
+
+    // returns true if the present value of a loan is zero
+    // true if all debt is repaid or debt is 100% written-off
+    function zeroPV(uint loan) public view returns (bool) {
+        return ((pile.debt(loan) == 0) || (pile.loanRates(loan) == WRITEOFF_RATE_GROUP));
+    }
+
+    // normalizes a timestamp to round down to the nearest midnight (UTC)
+    function uniqueDayTimestamp(uint timestamp) public pure returns (uint) {
```

Comment:
I think if the NAV is calcuated based on the `debt` which changes per second. It might make sense to remove the `uniqueDayTimestamp`  which basically enforces the NAV to be a step function per day. 

The NAV debt would increase only by day. Is this needed in this scenario for this version?